mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.java

```diff
@@ -1052,11 +1054,11 @@ private void writeResponse(final Http2Stream stream,
         throttledTransfer(response, socket, body, sink, body.size(), false);
         sink.close();
       } else if (response.isDuplex()) {
-        final BufferedSink sink = Okio.buffer(stream.getSink());
-        final BufferedSource source = Okio.buffer(stream.getSource());
-        final DuplexResponseBody duplexResponseBody = response.getDuplexResponseBody();
+        BufferedSink sink = Okio.buffer(stream.getSink());
```

Comment:
what there something wrong for them being `final`?

Reply:
`final` here won’t benefit the compiler and I claim that it harms readability. It’s using a lot of characters to say something that’s obvious.